vcd/resource_vcd_vm_insert_eject_media_test.go

```diff
@@ -0,0 +1,220 @@
+	return func(s *terraform.State) error {
+		injectItemRs, ok := s.RootModule().Resources[itemName]
+		if !ok {
+			return fmt.Errorf("not found: %s", itemName)
+		}
+
+		if injectItemRs.Primary.ID == "" {
+			return fmt.Errorf("no media insert ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*VCDClient)
+
+		_, vdc, err := conn.GetOrgAndVdc(testConfig.VCD.Org, testConfig.VCD.Vdc)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingVdcFromOrg, testConfig.VCD.Vdc, testConfig.VCD.Org, err)
+		}
+
+		vapp, err := vdc.FindVAppByName(vappNameForInsert)
+		if err != nil {
+			return err
+		}
+
+		vm, err := vdc.FindVMByName(vapp, vmNameForInsert)
+
+		if err != nil {
+			return err
+		}
+
+		for _, hardwareItem := range vm.VM.VirtualHardwareSection.Item {
+			if hardwareItem.ResourceSubType == "vmware.cdrom.iso" {
+				return nil
+			}
+		}
+		return fmt.Errorf("no media inserted found")
+	}
+}
+
+func testAccCheckMediaEjected(itemName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		injectItemRs, ok := s.RootModule().Resources[itemName]
+		if !ok {
+			return fmt.Errorf("not found: %s", itemName)
+		}
+
+		if injectItemRs.Primary.ID == "" {
+			return fmt.Errorf("no media insert ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*VCDClient)
+
+		_, vdc, err := conn.GetOrgAndVdc(testConfig.VCD.Org, testConfig.VCD.Vdc)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingVdcFromOrg, testConfig.VCD.Vdc, testConfig.VCD.Org, err)
+		}
+
+		vapp, err := vdc.FindVAppByName(vappNameForInsert)
+		if err != nil {
+			return err
+		}
+
+		vm, err := vdc.FindVMByName(vapp, vmNameForInsert)
+
+		if err != nil {
+			return err
+		}
+
+		for _, hardwareItem := range vm.VM.VirtualHardwareSection.Item {
+			if hardwareItem.ResourceSubType == "vmware.cdrom.iso" {
+				return nil
+			}
+		}
+		return fmt.Errorf("no media inserted found")
+	}
+}
+
+func testAccResourcesDestroyed(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*VCDClient)
+	for _, rs := range s.RootModule().Resources {
+		itemName := rs.Primary.Attributes["name"]
+		if rs.Type != "vcd_vm_insert_eject_media" && itemName != TestAccVcdMediaInsert {
+			continue
+		}
+
+		_, vdc, err := conn.GetOrgAndVdc(testConfig.VCD.Org, testConfig.VCD.Vdc)
+		if err != nil {
+			return fmt.Errorf(errorRetrievingVdcFromOrg, testConfig.VCD.Vdc, testConfig.VCD.Org, err)
+		}
+
+		_, err = vdc.FindVAppByName(vappNameForInsert)
+		if err != nil && !strings.Contains(err.Error(), "can't find vApp:") {
+			return fmt.Errorf("vapp %s still exist and error: %#v", itemName, err)
+		}
+
+		_, err = vdc.FindVDCNetwork(TestAccVcdVAppVmNetForInsert)
+		if err != nil && !strings.Contains(err.Error(), "can't find VDC Network:") {
+			return fmt.Errorf("network %s still exist and error: %#v", itemName, err)
+		}
+	}
+	return nil
+}
+
+const testAccCheckVcdInsertEjectBasic = `
+resource "vcd_network_routed" "{{.NetworkName}}" {
```

Comment:
One thought for experimenting with. We already have quite a bit of tests which create `vcd_network_routed`. You could try creating `vcd_network_isolated` instead. This would:
1.) Potentially make the test faster, as isolated network may be faster to create (needs double checking).
2.) Make the test simpler (would drop the need for edge gateway).
3.) We'd have more diversity in tests, which is always better to catch unexpected bugs.

Reply:
`network_isolated` is actually the slowest one to create.
The fastest one is `network_direct`.

Reply:
How peculiar! Let's keep routed then, as direct requires a privilege of having external IP addresses.

Reply:
we can't reuse cause network test will destroy created resources.